components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorSerializer.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.List;
+
+/**
+ * Created by riyafa on 7/10/17.
+ */
+public class CacheMediatorSerializer extends AbstractMediatorSerializer {
+
+    /**
+     * Stores certain parameters that are common to both Collector and Finder instances of the cache mediator
+     */
+    private CacheStore cacheStore;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected OMElement serializeSpecificMediator(Mediator mediator) {
+        if (!(mediator instanceof CacheMediator)) {
+            handleException("Unsupported mediator passed in for serialization : " + mediator.getType());
+        }
+
+        CacheMediator cacheMediator = (CacheMediator) mediator;
+        OMElement cacheElem = fac.createOMElement(CachingConstants.CACHE_LOCAL_NAME, synNS);
+        saveTracingState(cacheElem, mediator);
+
+        if (cacheMediator.getId() != null) {
+            cacheStore = CacheStoreManager.get(cacheMediator.getId());
+            cacheElem.addAttribute(fac.createOMAttribute("id", nullNS, cacheMediator.getId()));
+        } else {
+            cacheStore = CacheStoreManager.get("");
+        }
+
+        if (cacheMediator.isCollector()) {
+            cacheElem.addAttribute(fac.createOMAttribute("collector", nullNS, "true"));
+        } else {
+            cacheElem.addAttribute(fac.createOMAttribute("collector", nullNS, "false"));
+
+            if (cacheMediator.getTimeout() != 0) {
+                cacheElem.addAttribute(
+                        fac.createOMAttribute("timeout", nullNS, Long.toString(cacheMediator.getTimeout())));
+            }
+
+            if (cacheStore.getMaxMessageSize() != 0) {
+                cacheElem.addAttribute(
+                        fac.createOMAttribute("maxMessageSize", nullNS,
+                                              Integer.toString(cacheStore.getMaxMessageSize())));
+            }
+
+            OMElement onCacheHit;
+            if (cacheMediator.getOnCacheHitRef() != null) {
+                onCacheHit = fac.createOMElement("onCacheHit", synNS);
+                onCacheHit.addAttribute(
+                        fac.createOMAttribute("sequence", nullNS, cacheMediator.getOnCacheHitRef()));
+                cacheElem.addChild(onCacheHit);
+            } else if (cacheMediator.getOnCacheHitSequence() != null) {
+                onCacheHit = fac.createOMElement("onCacheHit", synNS);
+                serializeChildren(onCacheHit, cacheMediator.getOnCacheHitSequence().getList());
+                cacheElem.addChild(onCacheHit);
+            }
+
+            OMElement protocolElem = fac.createOMElement("protocol", synNS);
+            protocolElem.addAttribute(fac.createOMAttribute("type", nullNS, cacheStore.getProtocolType()));
+            if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(cacheStore.getProtocolType())) {
+
+                String[] methods = cacheStore.getHTTPMethodsToCache();
+                if (!(methods.length == 0 && "".equals(methods[0]))) {
```

Comment:
I am only checking the first element of an array. Do we need constants for this?